www/main.js

```diff
@@ -1,46 +1,48 @@
 ﻿$(function () {
     var roomArray = window.location.pathname.split('/');
-    roomName = roomArray[2];
+    var roomName = roomArray[2];
     var client = new Faye.Client('/faye');
 
     getLastUsername();
     getLastTheme();
 
     subscribeToRoom(client, roomName);
-    loadHistory();
+    loadHistory(roomName);
 
 
     setLastRoomName(roomName);
-    getRecentRoomNames();
+    getRecentRoomNames(roomName);
 
 
     $('#theme-picker').change(function () {
         setTheme(this.value);
     });
-    $('#input').keyup(function (e) {
-        var message = $('#input').val();
-        var userName = $('#name').val();
+
+    var $input = $('#input');
+    var $name = $('#name');
```

Comment:
Often a good idea to store jQuery elements in a variable rather than finding them with $('#id') every time. It's a bit quicker and it means you only need to update the selector in one place if it changes.